<a href="https://colab.research.google.com/github/gkmfrombs/Indian-Labor-Force-Education/blob/main/01_DataLoading_%26_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **📊 Exploratory Data Analysis: Indian Labor Force & Education**

**Dataset Source:** Periodic Labour Force Survey (PLFS) via NDAP (NITI Aayog)

**Core Objective:** Analyze how specific educational attainment levels impact the Unemployment Rate (UR) and Worker Population Ratio (WPR) across different Indian states and rural/urban demographics.

**🎯 Goals for this Notebook:**

Data Ingestion: Load the raw PLFS dataset and examine its initial structure and data types.

Data Cleaning: Handle any missing values, standardize column names, and address geographic edge cases (e.g., Chandigarh's purely urban classification).

Exploratory Visualizations: Uncover initial trends and correlations between education levels and employability to inform our future predictive models.

---

---
### ***[click here for the data](https://drive.google.com/file/d/1wk6Kkfatcx1KOylQfWHaJ68HQqJkEg8Y/view?usp=sharing)***

---



### Loading The Data
Necessary Libraries & Imports

In [ ]:
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [ ]:
df = pd.read_csv('/content/PLFS.csv')

### **EDA**

In [ ]:
print(df.columns)
df.shape

Index(['Country', 'State', 'Year', 'Type Of Areas', 'Gender',
       'Education Level',
       'Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1',
       'Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1',
       'Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1'],
      dtype='object')


(22650, 9)

In [ ]:
df.head()

,Country,State,Year,Type Of Areas,Gender,Education Level,"Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1","Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1","Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1"
0,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Not Literate,76.7,76.7,0.0
1,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Literate & Upto Primary,87.0,87.0,0.0
2,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Middle,89.7,88.0,1.9
3,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Secondary,58.8,58.8,0.0
4,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Higher Secondary,78.3,71.3,9.0


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22650 entries, 0 to 22649
Data columns (total 9 columns):
 #   Column                                                                                                                                     Non-Null Count  Dtype  
---  ------                                                                                                                                     --------------  -----  
 0   Country                                                                                                                                    22650 non-null  object 
 1   State                                                                                                                                      22650 non-null  object 
 2   Year                                                                                                                                       22650 non-null  object 
 3   Type Of Areas                                  

In [ ]:
print("missing values")
print("=================================")
df.isnull().sum()

missing values


,0
Country,0
State,0
Year,0
Type Of Areas,0
Gender,0
Education Level,0
"Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1",30
"Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1",30
"Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1",30


In [ ]:
df.describe()

,"Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1","Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1","Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1"
count,22620.000000,22620.000000,22620.000000
mean,56.570723,51.294892,9.059253
std,24.406334,23.137787,12.121258
min,0.000000,0.000000,0.000000
25%,39.475000,34.700000,1.000000
50%,58.700000,52.850000,5.200000
75%,75.800000,68.900000,12.200000
max,100.000000,100.000000,100.000000


In [ ]:
df.describe(include='object')

,Country,State,Year,Type Of Areas,Gender,Education Level
count,22650,22650,22650,22650,22650,22650
unique,1,36,7,3,3,10
top,India,Andaman and Nicobar Islands,"Year (Jul - Jun), 2023",Rural,Male,Not Literate
freq,22650,630,3240,7560,7560,2265


In [ ]:
cat_cols = df.select_dtypes(include='object').columns
for col in cat_cols:
    print(f"Unique values and their value counts in {col}:")
    print(df[col].unique(),df[col].value_counts())


Unique values and their value counts in Country:
['India'] Country
India    22650
Name: count, dtype: int64
Unique values and their value counts in State:
['Andaman and Nicobar Islands' 'Andhra Pradesh' 'Arunachal Pradesh'
 'Assam' 'Bihar' 'Chandigarh' 'Chhattisgarh' 'Delhi' 'Goa' 'Gujarat'
 'Haryana' 'Himachal Pradesh' 'Jammu and Kashmir' 'Jharkhand' 'Karnataka'
 'Kerala' 'Ladakh' 'Lakshadweep' 'Madhya Pradesh' 'Maharashtra' 'Manipur'
 'Meghalaya' 'Mizoram' 'Nagaland' 'Odisha' 'Puducherry' 'Punjab'
 'Rajasthan' 'Sikkim' 'Tamil Nadu' 'Telangana'
 'The Dadra and Nagar Haveli and Daman and Diu' 'Tripura' 'Uttar Pradesh'
 'Uttarakhand' 'West Bengal'] State
Andaman and Nicobar Islands                     630
Andhra Pradesh                                  630
Arunachal Pradesh                               630
Assam                                           630
Bihar                                           630
Chandigarh                                      630
Chhattisgarh              

### **Initial Data Insights**

Based on our initial exploration of the PLFS dataset, here are some key observations:

1.  **Dataset Size & Structure:**
    *   The dataset contains **22,650 rows** and **9 columns**, providing a substantial amount of information on the Indian labor force.
    *   Column names are highly descriptive but quite long, including units and scaling factors (e.g., `Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1`). These might benefit from shortening for easier analysis.

2.  **Missing Values:**
    *   There are **30 missing values** in the `Labor Force Participation Rate`, `Working Population Rate`, and `Unemployment Rate` columns. These will need to be handled during the data cleaning phase, possibly by imputation or removal.

3.  **Geographical Coverage:**
    *   The data covers **36 different States and Union Territories** across India.
    *   Most states have a consistent number of entries (630), but Lakshadweep has 620 and Puducherry has 610, indicating slight variations in data availability for these regions.

4.  **Temporal Coverage:**
    *   The dataset spans **7 years**, from `Year (Jul - Jun), 2017` to `Year (Jul - Jun), 2023`, allowing for analysis of trends over time.

5.  **Demographic and Area Breakdown:**
    *   Data is categorized by **'Rural'**, **'Urban'**, and **'Rural + Urban'** areas, and by **'Male'**, **'Female'**, and **'Persons'** (overall) for gender, enabling detailed demographic analysis.

6.  **Education Levels:**
    *   **10 distinct education levels** are present, ranging from 'Not Literate' to 'Post Graduate & Above', along with aggregated categories like 'Secondary & Above' and 'All'. This granular breakdown is crucial for our core objective of analyzing the impact of education.

7.  **Numerical Variable Summary:**
    *   **Labor Force Participation Rate (LFPR):** Ranges from 0% to 100%, with an average of approximately 56.6%.
    *   **Working Population Rate (WPR):** Also ranges from 0% to 100%, with an average of about 51.3%.
    *   **Unemployment Rate (UR):** Ranges from 0% to 100%, averaging around 9.1%. The maximum value of 100% suggests that there might be specific scenarios (e.g., very small population groups) or data points that need closer inspection.

    ---

In [ ]:
df.columns

Index(['Country', 'State', 'Year', 'Type Of Areas', 'Gender',
       'Education Level',
       'Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1',
       'Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1',
       'Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1'],
      dtype='object')

In [ ]:

# Renaming the columns to something manageable
rename_dict = {
    'Type Of Areas': 'Area_Type',
    'Education Level': 'Education_Level',
    'Labor Force Participation Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1': 'LFPR',
    'Working Population Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1': 'WPR',
    'Unemployment Rate According To Usual Status Based On Different General Education Level (UOM:%(Percentage)), Scaling Factor:1': 'UR'
}
df = df.rename(columns=rename_dict)
df.columns

Index(['Country', 'State', 'Year', 'Area_Type', 'Gender', 'Education_Level',
       'LFPR', 'WPR', 'UR'],
      dtype='object')

### **Data Visualization**

In [ ]:
# WE WILL START FROM HERE IN NEXT NOTEBOOK ...........................................................